pennylane_lightning/src/simulators/lightning_kokkos/observables/ObservablesKokkos.hpp

```diff
@@ -0,0 +1,380 @@
+     * @param obs Arguments to construct observables
+     */
+    static auto
+    create(std::initializer_list<PrecisionT> coeffs,
+           std::initializer_list<std::shared_ptr<Observable<StateVectorT>>> obs)
+        -> std::shared_ptr<Hamiltonian<StateVectorT>> {
+        return std::shared_ptr<Hamiltonian<StateVectorT>>(
+            new Hamiltonian<StateVectorT>{std::move(coeffs), std::move(obs)});
+    }
+
+    /**
+     * @brief Updates the statevector sv:->sv'.
+     * @param sv The statevector to update
+     */
+    void applyInPlace(StateVectorT &sv) const override {
+
+        StateVectorT buffer(sv.getNumQubits());
+        buffer.initZeros();
+
+        for (size_t term_idx = 0; term_idx < this->coeffs_.size(); term_idx++) {
+            StateVectorT tmp(sv);
+            this->obs_[term_idx]->applyInPlace(tmp);
+            LightningKokkos::Util::axpy_Kokkos<PrecisionT>(
+                ComplexT{this->coeffs_[term_idx], 0.0}, tmp.getView(),
+                buffer.getView(), tmp.getLength());
+        }
+        sv.updateData(buffer);
+    }
+};
+
+/**
+ * @brief A base class for all observable classes.
+ *
+ * We note that all subclasses must be immutable (does not provide any setter).
+ *
+ * @tparam T Floating point type
+ */
+
+template <typename T>
+class ObservableKokkos
+    : public std::enable_shared_from_this<ObservableKokkos<T>> {
+  private:
+    /**
+     * @brief Polymorphic function comparing this to another Observable
+     * object.
+     *
+     * @param Another instance of subclass of Observable<T> to compare
+     */
+    [[nodiscard]] virtual bool
+    isEqual(const ObservableKokkos<T> &other) const = 0;
+
+  protected:
+    ObservableKokkos() = default;
+    ObservableKokkos(const ObservableKokkos &) = default;
+    ObservableKokkos(ObservableKokkos &&) noexcept = default;
+    ObservableKokkos &operator=(const ObservableKokkos &) = default;
+    ObservableKokkos &operator=(ObservableKokkos &&) noexcept = default;
+
+  public:
+    virtual ~ObservableKokkos() = default;
+
+    /**
+     * @brief Apply the observable to the given statevector in place.
+     */
+    virtual void applyInPlace(StateVectorKokkos<T> &sv) const = 0;
+
+    /**
+     * @brief Get the name of the observable
+     */
+    [[nodiscard]] virtual auto getObsName() const -> std::string = 0;
+
+    /**
+     * @brief Get the wires the observable applies to.
+     */
+    [[nodiscard]] virtual auto getWires() const -> std::vector<size_t> = 0;
+
+    /**
+     * @brief Test whether this object is equal to another object
+     */
+    [[nodiscard]] bool operator==(const ObservableKokkos<T> &other) const {
+        return typeid(*this) == typeid(other) && isEqual(other);
+    }
+
+    /**
+     * @brief Test whether this object is different from another object.
+     */
+    [[nodiscard]] bool operator!=(const ObservableKokkos<T> &other) const {
+        return !(*this == other);
+    }
+};
```

Comment:
Perhaps a better approach would be to create an intermediate base for the sparse Hamiltonian and inherit here from the base.
I mean, following the same procedure we have in place for the other observables.

Reply:
I kept this for now, but I thought it would be simpler to have a sparse matrix interface to `expval` and `var` as in LQubit. I would thus remove `SparseHamiltonianKokkos` and `ObservableKokkos` altogether.